Mage/Archivist/Archivist.cs

```diff
@@ -240,35 +424,27 @@ public class Archivist : EventEmitter<JObject> {
 			return;
 		}
 
-		
-		// Default options
-
 
 		// Get data from server
-		rawMGet (realQueries, options, (Exception error, JToken result) => {
+		rawMGet (realQueries, options, (Exception error, JToken results) => {
 			if (error != null) {
 				cb (error, null);
 				return;
 			}
 
 			try {
-				JArray topicValues = (JArray)result;
-				foreach (JObject topicValue in topicValues) {
+				foreach (JObject topicValue in results as JArray) {
 					// Determine value cacheKeyName
 					string valueTopic = topicValue["key"]["topic"].ToString();
 					JObject valueIndex = (JObject)topicValue["key"]["index"];
-					string cacheKeyName = getCacheKey(valueTopic, valueIndex);
+					string cacheKeyName = GetCacheKey(valueTopic, valueIndex);
 
-					JToken responseResult = null;
-					if (topicValue["value"] != null) {
-						VaultValue newValue = new VaultValue(topicValue);
-						_cache.Add (cacheKeyName, newValue);
-						responseResult = (JToken)newValue.data;
-					}
+					// Set value to cache
+					SetCacheValue(topicValue);
 
 					// Add value to response
 					string responseKey = realQueryKeys[cacheKeyName];
-					responseObject.Add(responseKey, responseResult);
+					responseObject.Add(responseKey, GetCacheValue(cacheKeyName).data);
```

Comment:
SetCacheValue is called right above it so it should never return null I think, unless the expiration was set to 0?